server/pxf-service/src/test/java/org/greenplum/pxf/service/GPDataGenerator.java

```diff
@@ -0,0 +1,305 @@
+        row.add(String.format("{12345678900000.000001,12345678900000.000001}", rowIndex)); // 22 "numeric_arr  numeric[]"
+        row.add(String.format("{hello,world}", rowIndex));                                 // 23 "text_arr     text[]"
+        row.add(String.format("{11,12}", rowIndex));                                       // 24 "bytea_arr    bytea[]"
+        row.add(String.format("{abc,defij}", rowIndex));                                   // 25 "char_arr     bpchar(5)[]"
+        row.add(String.format("{abcde,fijkl}", rowIndex));                                 // 26 "varchar_arr  varchar(5)[]"
+
+        // overwrite null index
+        if (nullColumnIndex >=0) {
+            row.set(nullColumnIndex, null);
+        }
+        table.add(row);
+    }
+
+    /**
+     * Asserts that a given set of deserialized data represented by a collection of OneField objects contains all data
+     * in the sample dataset with correct values.
+     * @param rows a list of lists of OneField object - list of rows where each row is a list of column values
+     *             where each column value is a OneField object with type and value produced by a deserializer
+     */
+    public static void assertDataSet(List<List<OneField>> rows) {
+        // produce the sample dataset in memory and compare its size with the dataset provided
+        GPDataGenerator generator = new GPDataGenerator();
+        assertEquals(generator.table.size(), rows.size(), "Sizes of the datasets do not match");
+
+        // iterate over rows of the provided dataset
+        Iterator<List<Object>> tableIterator = generator.table.iterator();
+        int row = 0;
+        for (List<OneField> serializedRow : rows) {
+            List<Object> cells = tableIterator.next(); // pick a corresponding row from a sample dataset
+            // iterate over columns of a row of the provided dataset
+            int col = 0;
+            for (OneField field : serializedRow) {
+                // check the deserialized datatype
+                assertEquals(COLUMNS[col].getDeserializedType().getOID(), field.type, String.format("Type mismatch row=%d col=%d", row, col));
+                // check Java type / value
+                if (cells.get(col) == null) {
+                    assertNull(field.val, String.format("Expected null in row=%d col=%d", row, col));
+                } else {
+                    if (COLUMNS[col].getClazz() == ByteBuffer.class) {
+                        assertTrue(ByteBuffer.class.isAssignableFrom(field.val.getClass()));
+                    } else {
+                        assertSame(COLUMNS[col].getClazz(), field.val.getClass(), String.format("Java class mismatch row=%d col=%d", row, col));
+                    }
+                    // make adjustments to expected values
+                    Object expected = cells.get(col);
+                    if (col == 25) {
```

Comment:
i think having a comment here would be easier than having to scroll to find which col is col25 and why we need to add the trailing whitespace
```suggestion
                    if (col == 25) { \\ col 25 is charr_arr bpchar(5)[], we need to account for the trailing whitespace